nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/ParameterContextResource.java

```diff
@@ -400,11 +400,14 @@ public Response submitParameterContextUpdate(
     }
 
     private void validateParameterNames(final ParameterContextDTO parameterContextDto) {
-        for (final ParameterEntity entity : parameterContextDto.getParameters()) {
-            final String parameterName = entity.getParameter().getName();
-            if (!isLegalParameterName(parameterName)) {
-                throw new IllegalArgumentException("Request contains an illegal Parameter Name (" + parameterName + "). Parameter names may only include letters, numbers, spaces, and the special " +
-                    "characters .-_");
+        if (parameterContextDto.getParameters() != null) {
```

Comment:
Do we want to throw IllegalArgumentException if it's null?

Reply:
I don't know enough about the api.  It doesn't seem that there is a check that there _are_ parameters at all elsewhere.
Who is the sme?

Reply:
I'd vote for something like
````java
throw new IllegalArgumentException("Parameter Context must define at least one parameter.");
````

Reply:
ok

Reply:
Is there a way that needs to be expressed in the OpenAPI or the annotations?

Reply:
ok, so the parameters are not marked as being required

```java
@ApiModelProperty("The Parameters for the Parameter Context")
    public Set<ParameterEntity> getParameters() {
        return parameters;
    }
```

So I think that throwing here is against the openapi spec and the contract in general.
I'm not sure if chaffelson found this because of how openapi generated the code or not, but I don't think throwing here is right based on that.

Reply:
Parameters should not be required when creating a Parameter context. Currently you are expected to provide an empty list object, but it throws an NPE if parameters = null.
Either we should throw a reasonable error if the Parameters field is anything other than expected (rather than an NPE), or accept a null as if it was the empty list.

Reply:
So looking through the codebase at where `ParameterContextDTO.getParameters()` is called, about 50% of the time, we check if the value is `null`, and 50% of the time, we would throw a `NullPointerException`, unfortunately. This is one of those awkward coding situation, IMO, because *generally* I don't believe that code should return `null` values for collections - they should return an empty collection. But with DTO's, we generally want to return exactly what was given to it - there should be no logic such as "if null, use empty collection".

So I think either is perfectly acceptable. We can either say that the value MUST be populated and throw an IllegalArgumentException if it's not. OR we can say it's OK for that to be null, but in that situation we need to update everywhere in the code that looks at `ParameterContextDTO.getParameters()` - not just this Resource. I think I would lean toward the latter - allow a `null` value and just ensure that we always handle it properly. We should be lenient in what we permit.

Reply:
Couldn't we just make the getParameters call return an empty list if null or something and just change that spot?

Reply:
again, the the api doesn't have it as required, so accepting null i think is correct.